Setup.hs

```diff
@@ -1,30 +1,54 @@
 
 -- | Setup.hs module
 
+{-# OPTIONS_GHC -fno-warn-deprecations #-}
```

Comment:
According to the GHC documentation (various versions) this flag is called `warnings-deprecations` so you should use `-fno-warn-warnings-deprecations`. In fact, the synonymous name `deprecations` is not documented, and GHC shouldn't report something like

```
dist/setup/setup.hs:56:37: warning: [-Wdeprecations]
 ```

but

```
dist/setup/setup.hs:56:37: warning: [-Wwarnings-deprecations]
 ```

Please create a MVE a file an GHC issue.